src/Integration.UnitTests/Binding/CSharpVBBindingOperationTests.cs

```diff
@@ -396,13 +396,74 @@ public void ProjectBindingOperation_Prepare_NoAdditionalFileConflict_NoException
             additionalFileConflictChecker.VerifyAll();
         }
 
+        [TestMethod]
+        public void ProjectBindingOperation_Commit_AllRuleSetsAreDefault_AddsNonConditionalRuleSetProperty()
+        {
+            // Arrange
+            projectMock.SetCSProjectKind();
+            var testSubject = CreateTestSubject();
+            var debug = CreateRuleSetProperty(projectMock, "Debug", CSharpVBBindingOperation.DefaultProjectRuleSet);
+            var release = CreateRuleSetProperty(projectMock, "Release", CSharpVBBindingOperation.DefaultProjectRuleSet);
+
+            testSubject.Initialize();
+            testSubject.Prepare(CancellationToken.None);
+
+            var nonConditionalProperty = projectMock.GetBuildProperty(Constants.CodeAnalysisRuleSetPropertyKey);
+            nonConditionalProperty.Should().BeNullOrEmpty();
+
+            // Act
+            using (new AssertIgnoreScope()) // Ignore that the file is not on disk
+            {
+                testSubject.Commit();
+            }
+
+            // Assert
+            var solutionLevelRuleSet = PathHelper.CalculateRelativePath(testSubject.ProjectFullPath, cSharpVBBindingConfig.RuleSet.Path);
+
+            nonConditionalProperty = projectMock.GetBuildProperty(Constants.CodeAnalysisRuleSetPropertyKey);
+            nonConditionalProperty.Should().Be(solutionLevelRuleSet);
+            debug.Value.ToString().Should().Be(solutionLevelRuleSet);
+            release.Value.ToString().Should().Be(solutionLevelRuleSet);
+        }
+
+        [TestMethod]
+        public void ProjectBindingOperation_Commit_HasNonDefaultRuleSet_DoesNotAddNonConditionalRuleSetProperty()
+        {
+            // Arrange
+            projectMock.SetCSProjectKind();
+
+            var testSubject = CreateTestSubject();
+            var defaultRuleSet = CreateRuleSetProperty(projectMock, "Debug", CSharpVBBindingOperation.DefaultProjectRuleSet);
+            var nonDefaultRuleSet = CreateRuleSetProperty(projectMock, "Release", "myruleset.ruleset");
+
+            testSubject.Initialize();
+            testSubject.Prepare(CancellationToken.None);
+
+            // Act
+            using (new AssertIgnoreScope()) // Ignore that the file is not on disk
+            {
+                testSubject.Commit();
+            }
+
+            // Assert
+
+            var nonConditionalProperty = projectMock.GetBuildProperty(Constants.CodeAnalysisRuleSetPropertyKey);
+            nonConditionalProperty.Should().BeNullOrEmpty();
+
+            var solutionLevelRuleSet = PathHelper.CalculateRelativePath(testSubject.ProjectFullPath, cSharpVBBindingConfig.RuleSet.Path);
+            defaultRuleSet.Value.ToString().Should().Be(solutionLevelRuleSet);
+
+            var projectLevelRuleSet = Path.GetFileNameWithoutExtension(projectMock.FilePath) + ".Release.ruleset";
```

Comment:
I'm getting confused between the various changes and bug-fixes.
The test sets a non-default project-level ruleset for `Release` - is that correct?
If so, why are we expecting a new project-level ruleset to be created, rather than updating the existing one?

Reply:
Or is this fixed by your other PR (#1436)?

Reply:
Hmm, no, this is unfortunately looks like another bug in `CSharpVBBindingOperation`: 
https://github.com/SonarSource/sonarlint-visualstudio/blob/branch-furtherCM/src/Integration/Binding/CSharpVBBindingOperation.cs#L270

If the project references different rulesets, then we seem to wrap each ruleset with a new one :/
I'll open an issue for it.

Reply:
Opened an issue: https://github.com/SonarSource/sonarlint-visualstudio/issues/1437

Reply:
Is this a new issue? If it's a regression we should fix it before shipping, but if it's not new then it isn't essential.

Reply:
Not a regression, but I think it's a minor fix. I'll take a look at it and see if it's complicated/risky.